src/core/output-layout.cpp

```diff
@@ -1375,7 +1375,8 @@ class output_layout_t::impl
             auto& state  = entry.second;
             auto& lo     = this->outputs[handle];
 
-            if (!(state.source & OUTPUT_IMAGE_SOURCE_SELF))
+            if (!(state.source & OUTPUT_IMAGE_SOURCE_SELF ||
```

Comment:
This is not necessary if you look at the enum value definitions, I admit this is confusing. `DPMS` is nothing else but `SELF | OFF`, so that a dpms output is NOT removed here, even with the current master version.

Reply:
Oh I missed that. But weird it worked, not very well though. I've got a crash just now. I'll check the code more carefully.